radix-transaction-scenarios/src/scenarios/all_scenarios.rs

```diff
@@ -1,54 +1,422 @@
+                    .map_err(|_| ScenarioExecutorError::FailedToGetEpoch)?
+                    .as_typed::<VersionedConsensusManagerState>()
+                    .unwrap()
+                    .into_latest()
+                    .epoch;
+                let mut scenario = scenario_builder(ScenarioCore::new(
+                    self.network_definition.clone(),
+                    epoch,
+                    next_nonce,
+                ));
+                let metadata = scenario.metadata().clone();
+                (self.on_scenario_start)(&metadata);
+                let mut previous = None;
+                loop {
+                    let next = scenario
+                        .next(previous.as_ref())
+                        .map_err(|err| err.into_full(&scenario))
+                        .unwrap();
+                    match next {
+                        NextAction::Transaction(next) => {
+                            let receipt = self.execute_transaction(&next.raw_transaction)?;
+                            (self.on_transaction_execution)(
+                                &metadata,
+                                &next,
+                                &receipt,
+                                &self.database,
+                            );
+                            previous = Some(receipt);
+                        }
+                        NextAction::Completed(end_state) => {
+                            match self.nonce_handling {
+                                NonceHandling::Increment(increment) => next_nonce += increment,
+                                NonceHandling::StartAtNextAvailable => {
+                                    next_nonce = end_state.next_unused_nonce
+                                }
+                            }
+                            break;
+                        }
+                    }
+                }
             }
-            7 => Some(Box::new(|core| {
-                global_n_owned::GlobalNOwnedScenarioCreator::create(core)
-            })),
-            8 => Some(Box::new(|core| {
-                non_fungible_resource_with_remote_type::NonFungibleResourceWithRemoteTypeScenarioCreator::create(core)
-            })),
-            9 => Some(Box::new(|core| {
-                kv_store_with_remote_type::KVStoreScenarioCreator::create(core)
-            })),
-            10 => Some(Box::new(|core| {
-                max_transaction::MaxTransactionScenarioCreator::create(core)
-            })),
-            11 => Some(Box::new(|core| {
-                royalties::RoyaltiesScenarioCreator::create(core)
-            })),
-            _ => None,
         }
+
+        Ok(CompletedExecutionReceipt {
+            database: self.database,
+        })
+    }
+
+    fn execute_transaction(
+        &mut self,
+        transaction: &RawNotarizedTransaction,
+    ) -> Result<TransactionReceiptV1, ScenarioExecutorError> {
+        let network = NetworkDefinition::simulator();
```

Comment:
We should get the `network` from `self`.

Reply:
Yup, good point, I've made that change.